packages/std/src/errors/std_error.rs

```diff
@@ -0,0 +1,295 @@
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use snafu::Snafu;
+
+/// Structured error type for init, handle and query. This cannot be serialized to JSON, such that
+/// it is only available within the contract and its unit tests.
+///
+/// The prefix "Std" means "the standard error within the standard library". This is not the only
+/// result/error type in cosmwasm-std.
+#[derive(Debug, Serialize, Deserialize, Snafu, JsonSchema)]
+#[snafu(visibility = "pub(crate)")]
+#[serde(rename_all = "snake_case")]
+#[non_exhaustive]
+pub enum StdError {
```

Comment:
if it can not be serialized, how/why does it implement serde traits?

Reply:
This is an outdated doc comment. Thanks, will fix.

`StdError` can be serialized now. It just skips backtraces. During deserialization, those are set by the `Default` value, which is `None` for the optionals.

Reply:
awesome :)

Reply:
Done in 3a0b7eae0201bf9b60e3596d7e27ac0693074025. Plase feel free to raise further questions and challenge design such that we can either refine or communicate the reasoning better.

Reply:
I haven't reviewed the entire PR but i will add comments as i encounter point that i think can be improved :)